### ISSUE TITLE
Check if owns gmod when buying

### ISSUE THREAD

**Issue by Srlion** [everyday-as/gmodstore-issues#1060]
## Describe a feature you would like to have added to GmodStore
related to https://www.gmodstore.com/help/tickets/view/167552
It's really annoying that new made steam accounts can just buy without even owning gmod, I know that it was added before that users can only sign in if they have their profile set to public and it made issues.

The good way is to allow users to sign up but to buy addons you will have to get verified, and to verify

- Check when they sign up if their games are visible, if so then it's easy.
- Private? Then 2 ways to verify
  - Ask them to set their profile to public.
 OR
  - Ask for their steam api key https://steamcommunity.com/dev/apikey that can be used to check for gmod, and it will work because it's their steam api key

**Comment by WilliamVenner** [comment]
There is also the option of setting up a small Gmod server that can be used for verification.

```lua
hook.Add("CheckPassword", "Verify", function(steamid64)
    -- update database
    return false, "Account verified!"
end)
```

**Comment by WilliamVenner** [comment]
I don't think it would be too crazy of an idea to allow all three methods here. The user can make the choice of which one they want to do.

I'm not really sure if this issue is actually too much of an issue though, I personally wouldn't mind if a user hasn't verified that they own Gmod to buy my scripts, but that's coming from someone who has DRM.

**Comment by WilliamVenner** [comment]
Maybe the verification process should only be started when the user tries to checkout. Depends what we're most concerned about - losing customers on the registration page, or losing customers on the checkout page. The psychology here could be interesting. Is a user more likely to not be bothered at checkout or registration?

**Comment by Leystryku** [comment]
> There is also the option of setting up a small Gmod server that can be used for verification.
> 
> ```lua
> hook.Add("CheckPassword", "Verify", function(steamid64)
>     -- update database
>     return false, "Account verified!"
> end)
> ```
> 
> The only drawback of this is that this verification process could be hindered by a Gmod update. This can be easily solved with an auto updating script, though. The server could be shut down every 30 minutes to check for an update. To make the server start up as quickly as possible, use the x86-64 branch, the base gamemode and gm_flatgrass. That should start up pretty much instantly even on low end hardware.

you could also take the node workers from this project from ley and get a maybe 10mb standalone fake SRCDS instance which could handle the verification through the steam API https://github.com/Leystryku/LeySourceEngineProxyServ/tree/master/leyzyrenodeworker

**Comment by WilliamVenner** [comment]
> you could also take the node workers from this project from ley and get a maybe 10mb standalone fake SRCDS instance which could handle the verification through the steam API

If this could be used to fake the server's Gmod version as well, auto updating won't even be necessary :D

I think it's checked via A2S.

**Comment by JoshPiper** [comment]
> Maybe the verification process should only be started when the user tries to checkout. Depends what we're most concerned about - losing customers on the registration page, or losing customers on the checkout page. The psychology here could be interesting. Is a user more likely to not be bothered at checkout or registration?

It very much shouldn't be sprung on a user during checkout. You're likely to just have an abandoned cart at that point.
It could be done via an alert bubble, saying "you need to verify", or "you need to own gmod", and the cart should be disabled until that point.

With that being said, I don't think this is a good idea. We currently have creator accounts who don't own GMod, and this change would make it more difficult for businesses who need to separate personal and business accounts / expenses.
